libraries/chain/include/steem/chain/account_object.hpp

```diff
@@ -402,6 +403,25 @@ namespace steem { namespace chain {
    > change_recovery_account_request_index;
 } }
 
+#ifdef STEEM_ENABLE_SMT
+FC_REFLECT( steem::chain::account_object,
```

Comment:
Such reflection code should ifdef the specific fields rather than the entire reflection. This way if we have to change something else, it doesn't need to be updated in a second location.

Reply:
Absolutely, and that's how I coded it in initial commit, see https://github.com/steemit/steem/pull/2322/commits/251c9681709b9323d089c63337bfbb71d1225aa8
But it broke CI with following error in log:
```
error in steem::chain::account_object
doxygen: ['id', 'name', 'memo_key', 'json_metadata', 'proxy', 'last_account_update', 'created', 'mined', 'recovery_account', 'reset_account', 'last_account_recovery', 'comment_count', 'lifetime_vote_count', 'post_count', 'can_vote', 'voting_power', 'last_vote_time', 'balance', 'savings_balance', 'sbd_balance', 'sbd_seconds', 'sbd_seconds_last_update', 'sbd_last_interest_payment', 'savings_sbd_balance', 'savings_sbd_seconds', 'savings_sbd_seconds_last_update', 'savings_sbd_last_interest_payment', 'savings_withdraw_requests', 'reward_sbd_balance', 'reward_steem_balance', 'reward_vesting_balance', 'reward_vesting_steem', 'curation_rewards', 'posting_rewards', 'vesting_shares', 'delegated_vesting_shares', 'received_vesting_shares', 'vesting_withdraw_rate', 'next_vesting_withdrawal', 'withdrawn', 'to_withdraw', 'withdraw_routes', 'proxied_vsf_votes', 'witnesses_voted_for', 'last_post', 'last_root_post', 'post_bandwidth']
fc     : ['id', 'name', 'memo_key', 'json_metadata', 'proxy', 'last_account_update', 'created', 'mined', 'recovery_account', 'last_account_recovery', 'reset_account', 'comment_count', 'lifetime_vote_count', 'post_count', 'can_vote', 'voting_power', 'last_vote_time']
diff   : {'withdrawn', 'last_root_post', 'reward_steem_balance', 'withdraw_routes', 'savings_sbd_seconds_last_update', 'to_withdraw', 'delegated_vesting_shares', 'last_post', 'vesting_shares', 'vesting_withdraw_rate', 'savings_sbd_seconds', 'reward_vesting_balance', 'post_bandwidth', 'savings_balance', 'sbd_seconds_last_update', 'reward_vesting_steem', 'sbd_seconds', 'savings_withdraw_requests', 'balance', 'curation_rewards', 'savings_sbd_balance', 'next_vesting_withdrawal', 'witnesses_voted_for', 'savings_sbd_last_interest_payment', 'proxied_vsf_votes', 'sbd_balance', 'posting_rewards', 'received_vesting_shares', 'sbd_last_interest_payment', 'reward_sbd_balance'}
```
So I 'fixed' the problem with current solution that I don't like either for reasons you described.

Reply:
This code was reverted as part of other requested change.